### DISCUSSION TITLE
How to cancel a batch and all child batches within that?

### DISCUSSION THREAD

**Discussion by hineshamin** [sidekiq/sidekiq#6224] [Q&A]
I have a flow where I am creating child batches within child batches within child batches and am trying to figure out a good way to cancel all jobs easily. This is potentially how my batches would look like except there will be many more:

<img width="858" alt="Screenshot 2024-03-11 at 5 26 22 PM" src="https://github.com/sidekiq/sidekiq/assets/39443567/6f7a4988-11e1-4e5a-9336-68fbd2c4a1f3">

Through some testing I have seen that `batch.invalidate_all` called on batch 1 only applies to jobs within that batch (jobs 2, 3) and not through the child batches (2, 3, 4). So what is the best method to invalidating all jobs cascading down from a parent batch and not just the first layer.

**Comment by mperham** [comment]
You'd need to implement that yourself. Child batches can get access to their parent but parent batches do not have easy access to their children. You could create an overall "task id" which every job within the "pyramid of work" knows about. Each job would check if that task id has been cancelled.
